internal/secrets/secrets.go

```diff
@@ -0,0 +1,219 @@
+package secrets
+
+import (
+	"encoding/hex"
+	"encoding/json"
+	"os"
+	"os/exec"
+	"strings"
+
+	"github.com/pkg/errors"
+
+	"github.com/alecthomas/kingpin/v2"
+)
+
+type keyType int
+
+// Secret types.
+const (
+	Unset keyType = iota
+	Command
+	Config
+	EnvVar
+	File
+	Keychain
+	Value
+	Vault
+)
+
+// Secret holds secrets.
+type Secret struct {
+	Input      string
+	Value      string
+	StoreValue string
+	Type       keyType
+}
+
+// NewSecret constructs a new Secret.
+func NewSecret(value string) *Secret {
+	s := Secret{}
+
+	err := s.Set(value)
+	if err != nil {
+		return nil
+	}
+
+	return &s
+}
+
+// Set will set the Secret type.
+func (s *Secret) Set(value string) error {
+	switch {
+	case strings.HasPrefix(value, "envvar:"):
```

Comment:
repeating the same string twice is usually a recipe for copy/paste mistakes, better to user a map/slice and iterate it

Reply:
I prefer the switch, but I think I can kill 2 birds with 1 stone by defining the consts as the prefix, and using them here.